### ISSUE TITLE
v1.6.14 VM regression, compiler crash when accessing tuple field with enum set type using notin operator

### ISSUE THREAD

**Issue by jangko** [nim-lang/Nim#22190]
### Description

The following code will make the compiler crash

```Nim
type
  EVMFork = enum
    Berlin
    Istanbul
    Shanghai

const
  Vm2OpAllForks =
    {EVMFork.low .. EVMFork.high}

  vm2OpExecBlockData = [(forks: Vm2OpAllForks)]

proc mkOpTable(selected: EVMFork): bool =
  selected notin vm2OpExecBlockData[0].forks

const
  tab = mkOpTable(Berlin)
```

### Nim Version

Nim Compiler Version 1.6.14 [Windows: amd64]
Compiled at 2023-06-29
Copyright (c) 2006-2023 by Andreas Rumpf

git hash: 71ba2e7f3c5815d956b1ae0341b0743242b8fec6
active boot switches: -d:release

### Current Output

```text
..........................................................fatal.nim(54)            sysFatal
Error: unhandled exception: nimsets.nim(16, 10) `s.kind == nkCurly`  [AssertionDefect]
```

### Expected Output

```text
no error
```

### Additional Information

it works with v1.6.12 and v1.9.5 and can compile using `let tab = mkOpTable(Berlin)`.

**Comment by juancarlospaco** [comment]
@github-actions nim c

```nim
type EVMFork = enum Berlin, Istanbul, Shanghai
const Vm2OpAllForks = {EVMFork.low .. EVMFork.high}
const vm2OpExecBlockData = [(forks: Vm2OpAllForks)]
proc mkOpTable(selected: EVMFork): bool =
  selected notin vm2OpExecBlockData[0].forks
const tab = mkOpTable(Berlin)
```

**Comment by al6x** [comment]
It still happens, v 1.6.14

**Comment by ringabout** [comment]
It is not fixed though.

**Comment by narimiran** [comment]
My bisecting on the 1.6 branch showed that this is caused by https://github.com/nim-lang/Nim/pull/21318. :/
I wasn't able to find a commit in `devel` which fixes this.

(cc @kdeme)

**Comment by ringabout** [comment]
You can simply revert https://github.com/nim-lang/Nim/pull/21318

**Comment by juancarlospaco** [comment]
!nim c

```nim
type EVMFork = enum Berlin, Istanbul, Shanghai
const Vm2OpAllForks = {EVMFork.low .. EVMFork.high}
const vm2OpExecBlockData = [(forks: Vm2OpAllForks)]
proc mkOpTable(selected: EVMFork): bool =
  selected notin vm2OpExecBlockData[0].forks
const tab = mkOpTable(Berlin)
```

**Comment by juancarlospaco** [comment]
Works on my machine on devel.

**Comment by narimiran** [comment]
> You can simply revert https://github.com/nim-lang/Nim/pull/21318

Done.

@jangko, please test and see if this can be closed.

**Comment by jangko** [comment]
@narimiran okay

**Comment by jangko** [comment]
@narimiran get another regression:
```
Error: invalid pragma: optimization: speed
```

```Nim
const
  lowmem {.intdefine.}: int = 0
  lowMemoryCompileTime {.used.} = lowmem > 0

proc selectVM() =
  while true:
    when not lowMemoryCompileTime:
      when defined(release):
        when defined(windows):
          when defined(cpu64):
            {.warning: "*** Win64/VM2 handler switch => computedGoto".}
            {.computedGoto, optimization: speed.}
          else:
            # computedGoto not compiling on github/ci (out of memory) -- jordan
            {.warning: "*** Win32/VM2 handler switch => optimisation disabled".}
            # {.computedGoto, optimization: speed.}

        elif defined(linux):
          when defined(cpu64):
            {.warning: "*** Linux64/VM2 handler switch => computedGoto".}
            {.computedGoto, optimization: speed.}
          else:
            {.warning: "*** Linux32/VM2 handler switch => computedGoto".}
            {.computedGoto, optimization: speed.}

        elif defined(macosx):
          when defined(cpu64):
            {.warning: "*** MacOs64/VM2 handler switch => computedGoto".}
            {.computedGoto, optimization: speed.}
          else:
            {.warning: "*** MacOs32/VM2 handler switch => computedGoto".}
            {.computedGoto, optimization: speed.}

        else:
          {.warning: "*** Unsupported OS => no handler switch optimisation".}
```

**Comment by narimiran** [comment]
> @narimiran get another regression:

I cannot reproduce the bug on my machine with the example you provided.

**Comment by jangko** [comment]
> I cannot reproduce the bug on my machine with the example you provided.

Sorry, the previous example is incomplete. I have edited it.

run with this compiler switch, assume the file name is `speed.nim`:
```text
nim c -d:release speed
```

The error happen on both windows and linux amd64.

**Comment by narimiran** [comment]
> Sorry, the previous example is incomplete. I have edited it.

Yes, now I can reproduce it.

Bisecting shows it is this PR's fault: https://github.com/nim-lang/Nim/pull/19646 (cc @Araq)

**Comment by jangko** [comment]
Aside from the `optimization: speed`, I think this issue can be closed. The `optimization` pragma issue seems not like a bug or regression. I will patch nimbus-eth1 when nim 1.6.16 is ready.

**Comment by narimiran** [comment]
>  I will patch nimbus-eth1 when nim 1.6.16 is ready.

Could you please test `nimbus-eth1` with the current `HEAD` of the `version-1-6` branch, so if there is any other regression, we can also fix those for 1.6.16?

cc @kdeme

**Comment by jangko** [comment]
Hai @narimiran , I've tested `nimbus-eth1` with current `HEAD` of `version-1-6` branch. No more regression. Looks good.
